src/App.tsx

```diff
@@ -1,51 +1,64 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
+
 import './App.scss';
+import { SortType } from './enums/sort-type.enum';
+import { Good } from './types/Good.type';
+import { goodsFromServer } from './data/goods-from-server';
+import { GoodList } from './components/GoodList';
+import { ButtonGroup } from './components/ButtonGroup';
+
+function getPreparedGoods(
+  goods: Good[],
+  sortField: SortType | '',
+  isReversed: boolean,
+): Good[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SortType.BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+
+  const reset = () => {
```

Comment:
The function name `reset` is not starting with a verb which makes it less descriptive of the action it performs. Consider renaming it to `handleReset` or `resetSorting` to comply with the [CODE STYLE] checklist item.

Reply:
ok